### PR TITLE
More GL state leak fixing

### PR DESCRIPTION
**Pull request by hrydgard**
Might help #13157 if we're lucky, but I still don't see how #13295 could have broken it like that.

### CODE REVIEW

File: ext/native/thin3d/thin3d_gl.cpp [hunk 1]
```diff
@@ -1037,6 +1047,8 @@ void OpenGLContext::ApplySamplers() {
 		GLenum magFilt = samp->magFilt;
 		GLenum minFilt = tex->HasMips() ? samp->mipMinFilt : samp->minFilt;
 		renderManager_.SetTextureSampler(i, wrapS, wrapT, magFilt, minFilt, 0.0f);
+		// TODO: Improve this to allow mipmaps. We don't care about those right now though for thin3d stuff.
```
Comment (unknownbrackets) [inline ext/native/thin3d/thin3d_gl.cpp:1050]: Don't we use generated mipmaps for screenshots in the UI?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]

**Comment by Gamemulatorer** [comment]
It wasn't broke by #13295
> issue started in [9ebb402](https://github.com/hrydgard/ppsspp/commit/9ebb402e4fa9900a9b94ed64814c4386e50c0437) last working build no screen flickering [bb6219e](https://github.com/hrydgard/ppsspp/commit/bb6219e40202383f38f3439d7a3d78e2389024c9)

**Comment by hrydgard** [comment]
Looking at the sequence of changes between those builds on the buildbot, #13295 is the only thing that stands out as suspicious to me (though I don't quite understand it)

**Comment by Gamemulatorer** [comment]
Ok I wait for this to be merge and then test the buildbot apk..

**Comment by Gamemulatorer** [comment]
Confirmed this pr fix #13157

**Comment by hrydgard** [comment]
Thanks for testing @Gamemulatorer !

File: ext/native/thin3d/thin3d_gl.cpp (hunk 1 above)
Comment (hrydgard) [reply ext/native/thin3d/thin3d_gl.cpp]: I fixed this properly.

**Comment by Gamemulatorer** [comment]
Why this pr causing the crashes of #13307

**Comment by hrydgard** [comment]
Because I added a check for something I didn't think could happen, but it did. I'm gonna fix it up.

**Pull request merged**